packages/webpack/test/fixtures/main/main.js

```diff
@@ -2,12 +2,17 @@ import { hello as helloCommonJS } from 'commonjs-package'
 import { re } from 'commonjs-quirks'
 import { hello as helloES6 } from 'es6-module-package'
 import globals from 'globals-package'
+import 'resource-hack'
 import 'side-effects-package/styles.css'
 import { hello as helloTypeScript } from 'typescript-package'
 import { hello as helloUMD } from 'umd-package'
 import { what } from './src/hello.ts'
 import { thing } from './src/other.mjs'
 
+// This is a webpack feature where the file referenced here is included in the
+// dist as an asset. LavaMoat plugin allows this in app, but not in packages.
+const x = new URL('./src/ambientasset.html', import.meta.url)
```

Comment:
How does this feature work under the hood? Is URL overwritten by webpack? How does webpack know to take these when they're invoked and do its voodoo?

Reply:
I'd be curious to know how prevalent this sort of thing is in 3rd-party packages.

Reply:
Webpack extracts this at compile time (I think without running the code, just from AST) and does its best to ensure the file is included in dist for when you put it in a link or use as img src.

I've never used it, but it seems to be in use unfortunately.